snap/local/hooks/cmd/configure/configure.go

```diff
@@ -99,4 +101,10 @@ func main() {
 		hooks.Error(fmt.Sprintf("Invalid value for 'autostart' : %s", autostart))
 		os.Exit(1)
 	}
+
+	log.SetComponentName("configure")
+	if err := options.ProcessAppConfig("device-mqtt"); err != nil {
+		hooks.Error(fmt.Sprintf("could not process options: %v", err))
+		os.Exit(1)
+	}
```

Comment:
With the new [version](https://github.com/canonical/edgex-snap-hooks/pull/41), this should be called before processing the lagacy `env.` options, i.e. to line 53.